src/game/common/system/asciistring.cpp

```diff
@@ -21,56 +21,73 @@
 
 Utf8String const Utf8String::s_emptyString(nullptr);
 
-Utf8String::Utf8String() :
-    m_data(nullptr)
+/**
+ * Default constructor added for convenience
+ */
+Utf8String::Utf8String() : m_data(nullptr)
 {
 }
 
-Utf8String::Utf8String(const char *s) :
-    m_data(nullptr)
+/**
+ * Inits this string with an existing string (copy) and increments reference count
+ */
+Utf8String::Utf8String(Utf8String const &string) : m_data(string.m_data)
+{
+    if (m_data != nullptr) {
+        m_data->Inc_Ref_Count();
+    }
+}
+
+/**
+ * Inits this string with a reference to the start of a char array.
+ */
+Utf8String::Utf8String(const char *s) : m_data(nullptr)
 {
     if ( s != nullptr ) {
+		//Get length of the string that was passed
```

Comment:
Do we have tabs here instead of spaces? Alignment looks odd in the github diff. The whole file looks like needs the current formatting rules applied as its predates the clang-format file and is inconsistent with more recent code.

Reply:
As discussed i will add a commit to do all the Clang stuff.